move-mutator/src/operators/binary_swap.rs

```diff
@@ -33,22 +36,47 @@ impl BinarySwap {
 
 impl MutationOperator for BinarySwap {
     fn apply(&self, source: &str) -> Vec<MutantInfo> {
-        // There is no point in swapping the operator for these cases as it would result in the same expression.
-        if self.operation == Operation::Add
-            || self.operation == Operation::Mul
-            || self.operation == Operation::Eq
-            || self.operation == Operation::Neq
-        {
+        // Check if we've exactly two expressions.
+        let exps_len = self.exps.len();
+        if exps_len != 2 {
+            warn!("BinarySwapOperator: Expected exactly two expressions, got {exps_len}");
             return vec![];
         }
 
-        // Check if we've exactly two expressions.
-        if self.exps.len() != 2 {
-            warn!(
-                "BinarySwapOperator: Expected exactly two expressions, got {}",
-                self.exps.len()
-            );
-            return vec![];
+        // There is no point in swapping the operator for some cases, as it would result in the same expression.
```

Comment:
Mention that:
For any commutative binary operator (i.e., `a op b == b op a`), when the operands are pure, the binary swap mutation results in equivalent code.

You could even create an `is_commutative` helper function to clarify what's happening.

Reply:
Done